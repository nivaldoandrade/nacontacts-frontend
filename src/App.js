import { ThemeProvider } from 'styled-components';

import theme from './assets/styles/themes/default';

import GlobalStyles from './assets/styles/global';

function App() {
  return (
    <ThemeProvider theme={theme}>
      <h1>Nacontacts</h1>
      <GlobalStyles />
    </ThemeProvider>
  );
}

export default App;
