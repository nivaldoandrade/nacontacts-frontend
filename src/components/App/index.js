import { ThemeProvider } from 'styled-components';

import theme from '../../assets/styles/themes/default';

import GlobalStyles from '../../assets/styles/global';
import { Container } from './styles';
import { Header } from '../Header';
// import { ContactList } from '../ContactsList';
import { Routes } from '../../routes';

function App() {
  return (
    <ThemeProvider theme={theme}>
      <Container>
        <Header />
        {/* <ContactList /> */}
        <Routes />
      </Container>
      <GlobalStyles />
    </ThemeProvider>
  );
}

export default App;
