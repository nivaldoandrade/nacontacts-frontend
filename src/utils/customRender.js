/* eslint-disable import/no-extraneous-dependencies */
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import theme from '../assets/styles/themes/default';

export const customRender = (ui, options) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options);
