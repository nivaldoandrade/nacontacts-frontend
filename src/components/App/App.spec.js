import { act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import App from '.';
import { customRender } from '../../utils/customRender';

describe('App Componenent', () => {
  it('renders correctly', async () => {
    const { container } = customRender(<App />, { wrapper: BrowserRouter });

    await act(() => {
      expect(container).toMatchSnapshot();
    });
  });
});
