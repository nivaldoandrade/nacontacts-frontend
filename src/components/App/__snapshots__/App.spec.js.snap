// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Componenent renders correctly 1`] = `
<div>
  <div
    class="sc-bcXHqe lbzehy"
  >
    <header
      class="sc-gswNZR gAmjSf"
    >
      <img
        alt="NaContacts Logo"
      />
    </header>
    <div
      class="sc-kDvujY eKLFux"
    >
      <header
        class="sc-ftTHYK cgWILH"
      >
        <a
          href="/new"
        >
          Novo Contato
        </a>
      </header>
      <div
        class="sc-ipEyDJ kSvPRz"
      />
      <div
        class="sc-eDWCr jcpntM"
      />
    </div>
  </div>
  <div
    class="sc-hHTYSt kRtauP"
  />
</div>
`;
